jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -0,0 +1,109 @@
+package ml4ir.inference.tensorflow
+
+import scala.collection.JavaConverters._
+import ml4ir.inference.tensorflow.data.{ModelFeaturesConfig, StringMapExampleBuilder, TestData}
+import org.junit.Test
+import org.junit.Assert._
+import org.tensorflow.example._
+
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+@Test
+class TensorFlowInferenceIT extends TestData {
+  val classLoader = getClass.getClassLoader
+
+  def resourceFor(path: String) = classLoader.getResource(path).getPath
+
+  /**
+   * Class holding a Example and prediction from Python
+   * @param sequence the Example context
+   * @param scores the prediction scores
+   */
+  class PredictionVector(var sequence: Map[String, String], var scores: Array[Float])
+
+  /**
+   * Ugly parsing of the Python CSV predicition.
+   *
+   * Mainly removing the "b'" formatting and slicing into List and Map.
```

Comment:
so do we know that the serialization that python uses allows us to just pull those b"blah" stuff out directly?  I think they're perhaps "UTF-8" byte-strings?

Reply:
Here's a snippet:

> query_key,query_text,query_words,domain_id,user_context,code,entity_id,category_label
> b'query_id_0',b'bernardo and unfold',"[b'bernardo' b'and' b'unfold' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b''
>  b'' b'' b'' b'' b'' b'']",b'S',"[b'DDD' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b''
>  b'' b'' b'']",b'canary',[0. 0. 1. 0. 0. 0. 0. 0. 0.],"[2.9891068e-01 2.4615353e-01 2.2974105e-01 8.6946599e-02 1.3798673e-01
>  5.1893934e-05 6.1736333e-05 1.2362415e-04 2.4173807e-05]"
> b'query_id_1',b'stand elsinore prince am yourself make twelve',"[b'stand' b'elsinore' b'prince' b'am' b'yourself' b'make' b'twelve' b''
>  b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'' b'']",b'L',"[b'EEE' b'EEE' b'BBB' b'CCC' b'BBB' b'BBB' b'AAA' b'EEE' b'GGG' b'FFF'
>  b'CCC' b'GGG' b'GGG' b'' b'' b'' b'' b'' b'' b'']",b'canary',[0. 0. 0. 0. 1. 0. 0. 0. 0.],"[3.0492017e-01 1.9346356e-01 2.2024015e-01 1.2455909e-01 1.5630856e-01
>  9.8194840e-05 1.1984430e-04 2.1972452e-04 7.0748545e-05]"

It's numpy.to_csv I think. This adds the extra typing stuff around the CSV value (like the bracket to indicate a list etc...).
I've look to change this output, I'm not sure there is an easy way.

Reply:
we should make a note (followup story) to pick a better serialization format for that CSV.  Maybe store it in JSON, at the very minimum, and without python-isms like the b'foo' stuff.

Reply:
It may be addressable in this review.

Initially I wrote a standalone script that was calling the `predict` method with a param to log prediction in an external log files. For what I can remember the output was cleaner than that. Be we lose the benefit of not having to rely on an other script to generate a model.

Or I can try to fix the output.csv in Python.

Reply:
@balikasg / @lastmansleeping do we programmatically (in python) read the output.csv anywhere?  Or is it just for debugging / humans?

Reply:
Not for ranking at least. It is just for debugging and error analysis (which we haven't really done much of). I remember the QU team considering reading the output file to compute further metrics in a spark job. Not sure if that fruitioned though.

Reply:
if they want to read it in a spark job, definitely want it in a better format than this.

Alain - looks like maybe changing the format from the python side, might be best?

Reply:
this is how bytes are saved in a CSV
For python, simply calling `ast.literal_eval` on the column fixes the issue. We can fix this before saving the csv if it is a problem, I see in scala it needs a hack..

Reply:
and yes, we are using this output exactly as @lastmansleeping said!

Reply:
WE should make the change in Python side, yes. I will try to raise a PR today

Reply:
@balikasg I can address this in this current PR I think.

Reply:
I think it would be best to open a separate PR for this